llama_stack/providers/utils/memory/vector_store.py

```diff
@@ -167,7 +167,11 @@ async def add_chunks(self, chunks: list[Chunk], embeddings: NDArray):
         raise NotImplementedError()
 
     @abstractmethod
-    async def query(self, embedding: NDArray, k: int, score_threshold: float) -> QueryChunksResponse:
+    async def query_vector(self, embedding: NDArray, k: int, score_threshold: float) -> QueryChunksResponse:
+        raise NotImplementedError()
+
+    @abstractmethod
+    async def query_keyword(self, query_string: str | None, k: int, score_threshold: float) -> QueryChunksResponse:
```

Comment:
```suggestion
        query_string: Optional[str],
```

Reply:
Ruff is reverting the change back to use `str|None` in order to comply with Python 3.10+ PEP 0604 proposal.